source/library/HW/Worker.hs

```diff
@@ -0,0 +1,136 @@
+module HW.Worker (worker) where
+
+import qualified CMark
+import qualified Control.Concurrent as Concurrent
+import qualified Control.Monad as Monad
+import qualified Control.Monad.Trans.Reader as Reader
+import qualified Data.Aeson as Aeson
+import qualified Data.IORef as IORef
+import qualified Data.List as List
+import qualified Data.List.NonEmpty as NonEmpty
+import qualified Data.Map as Map
+import qualified Data.Maybe as Maybe
+import qualified Data.Ord as Ord
+import qualified Data.Text as Text
+import qualified Data.Text.Encoding as Encoding
+import qualified Data.Time as Time
+import qualified HW.Handler.Issue as Issue
+import qualified HW.Type.Config as Config
+import qualified HW.Type.Date as Date
+import qualified HW.Type.Issue as Issue
+import qualified HW.Type.Listmonk as Listmonk
+import qualified HW.Type.Number as Number
+import qualified HW.Type.State as State
+import qualified Network.HTTP.Client as Client
+import qualified Network.HTTP.Types as Http
+
+worker :: IORef.IORef State.State -> IO ()
+worker stateRef = Monad.forever $ do
+  now <- Time.getCurrentTime
+  putStrLn $ "[worker] running at " <> Time.formatTime Time.defaultTimeLocale "%Y-%m-%dT%H:%M:%SZ" now
+  state <- IORef.readIORef stateRef
+  case Config.listmonk $ State.config state of
+    Nothing -> putStrLn "[worker] missing listmonk config"
+    Just listmonk -> do
+      let maybeLatest =
+            Maybe.listToMaybe
+              . List.sortOn (Ord.Down . Issue.issueDate)
+              . Map.elems
+              $ State.issues state
+      case maybeLatest of
+        Nothing -> putStrLn "[worker] no issues"
+        Just issue -> do
+          let url = Listmonk.url listmonk
+          getRequest <- Client.parseRequest . Text.unpack $ url <> "/api/campaigns"
+          let username = Encoding.encodeUtf8 $ Listmonk.username listmonk
+          let password = Encoding.encodeUtf8 $ Listmonk.password listmonk
+          let name = "hwn-" <> Date.toShortText (Issue.issueDate issue)
+          let query = Http.renderQuery True [("query", Just $ Encoding.encodeUtf8 name)]
```

Comment:
I should add a comment explaining that this `query` parameter is dropped into a SQL `LIKE` expression, like this:

```
...?query=foo
-- becomes
... LIKE '%foo%'
```

That's why I search for `hwn-YYYY-MM-DD` rather than the issue number. Something like `hwn-issue-12` could also match `hwn-issue-123`, which is not desired.

Reply:
```suggestion
          -- `?query=x` becomes `like '%x%'`, so this will match any substring.
          -- That's why we search for `hwn-YYYY-MM-DD` rather than `issue-NNN`.
          let query = Http.renderQuery True [("query", Just $ Encoding.encodeUtf8 name)]
```